backend/src/common/util/workflow.go

```diff
@@ -64,7 +70,12 @@ func (w *Workflow) OverrideParameters(desiredParams map[string]string) {
 func (w *Workflow) VerifyParameters(desiredParams map[string]string) error {
 	templateParamsMap := make(map[string]*string)
 	for _, param := range w.Spec.Arguments.Parameters {
-		templateParamsMap[param.Name] = param.Value
+		var val string
+		if param.Value != nil {
+			val = param.Value.String()
```

Comment:
reading this file, it seems we need a
- `*string` to `*AnyString` and also an `*AnyString` to `*string` conversion util

Reply:
added this.